From 43f56de6ba7606f8e18ac3da38091a9f3ab6741c Mon Sep 17 00:00:00 2001
From: Peng Ma <peng.ma@nxp.com>
Date: Thu, 26 Jul 2018 08:01:49 +0000
Subject: [PATCH 058/741] dmaengine: fsl-qdma: remove remaining code for SG
 mode

commit c4ab2b3b80147fa43c397d90d5719c42fb01bc87 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

remove remaining code for SG mode to clean up the qdma driver

Signed-off-by: Peng Ma <peng.ma@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/dma/fsl-qdma.c | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/drivers/dma/fsl-qdma.c b/drivers/dma/fsl-qdma.c
index 79fa4a4637a3..b5960abc44da 100644
--- a/drivers/dma/fsl-qdma.c
+++ b/drivers/dma/fsl-qdma.c
@@ -516,45 +516,7 @@ static struct fsl_qdma_queue
 			queue_temp->virt_head = queue_temp->cq;
 	 		queue_temp->virt_tail = queue_temp->cq;
          
-			/*
-                         * The dma pool for queue command buffer
-                         */
-                        queue_temp->comp_pool =
-                        dma_pool_create("comp_pool",
-                                      &pdev->dev,
-                                      FSL_QDMA_BASE_BUFFER_SIZE,
-                                      16, 0);
-                        if (!queue_temp->comp_pool) {
-                                dma_free_coherent(&pdev->dev,
-                                         sizeof(struct fsl_qdma_format) *
-                                         queue_size[i],
-                                         queue_temp->cq,
-                                         queue_temp->bus_addr);
-                                return NULL;
-                        }
-                        /*
-                         * The dma pool for queue command buffer
-                         */
-                        queue_temp->sg_pool =
-                        dma_pool_create("sg_pool",
-                                       &pdev->dev,
-                                       FSL_QDMA_EXPECT_SG_ENTRY_NUM * 16,
-                                       64, 0);
-                        if (!queue_temp->sg_pool) {
-                                dma_free_coherent(&pdev->dev,
-                                         sizeof(struct fsl_qdma_format) *
-                                         queue_size[i],
-                                         queue_temp->cq,
-                                         queue_temp->bus_addr);
-                                dma_pool_destroy(queue_temp->comp_pool);
-                                return NULL;
-                        }
-
-			/*
-			 * List for queue command buffer
-			 */
 			INIT_LIST_HEAD(&queue_temp->comp_used);
-			INIT_LIST_HEAD(&queue_temp->comp_free);
 			spin_lock_init(&queue_temp->queue_lock);
 		}
 	}
-- 
2.17.1

